src/cookbook/packaging-components-npm.md

```diff
@@ -0,0 +1,197 @@
+
+```html
+<script src="https://unpkg.com/vue"></script>
+<script src="https://unpkg.com/my-component"></script>
+...
+<my-component></my-component>
+...
+```
+
+Not only does this help you avoid copy/pasting components around, but it also allows you to give back to the Vue community!
+
+## Can't I Just Share `.vue` Files Directly?
+
+Vue already allows components to be written as a single file. Because a Single File Component (SFC) is already just one file, you might ask:
+
+> "Why can't people use my `.vue` file directly? Isn't that the simplest way to share components?"
+
+It's true, you can share `.vue` files directly, and anyone using a [Vue build](/installation.html#explanation-of-different-builds) containing the Vue compiler can consume it immediately. Also, the SSR build uses string concatenation as an optimization, so the `.vue` file might be preferred in this scenario (see [Packaging Components for npm > SSR Usage](#SSR-Usage) for details). However, this excludes anyone who wishes to use the component directly in a browser via `<script>` tag, anyone who uses a runtime-only build, or build processes which don't understand what to do with `.vue` files.
+
+Properly packaging your SFC for distribution via npm enables your component to be shared in a way which is ready to use everywhere!
+
+## Packaging Components for npm
+
+For the purposes of this section, assume the following file structure:
+
+```
+package.json
+build/
+   rollup.config.js
+src/
+   wrapper.js
+   my-component.vue
+dist/
+```
+
+### How does npm know which version to serve to a browser/build process?
+
+The `package.json` file used by npm really only requires one version (`main`), but as it turns out, we aren't limited to that. We can address the most common use cases by specifying 2 additional versions (`module` and `unpkg`), and provide access to the `.vue` file itself using the `browser` field. A sample package.json would look like this:
+
+```json
+{
+  "name": "my-component",
+  "version": "1.2.3",
+  "main": "dist/my-component.umd.js",
+  "module": "dist/my-component.esm.js",
+  "unpkg": "dist/my-component.min.js",
+  "browser": {
+    "./sfc": "src/my-component.vue"
+  },
+  ...
+}
+```
+
+When build tools are used, they will pick up on the `module` build. Legacy applications would use the `main` build, and the `unpkg` build can be used directly in browsers. In fact, the [unpkg](https://unpkg.com) cdn automatically uses this when someone enters the URL for your module into their service!
+
+### SSR Usage
+
+You might have noticed something interesting - browsers aren't going to be using the `browser` version. That's because this field is actually intended to allow authors to provide [hints to bundlers](https://github.com/defunctzombie/package-browser-field-spec#spec) which in turn create their own packages for client side use. With a little creativity, this field allows us to map an alias to the `.vue` file itself. For example:
```

Comment:
```suggestion
You might have noticed something interesting - browsers aren't going to be using the `browser` version. That's because this field is actually intended to allow authors to provide [hints to bundlers](https://github.com/defunctzombie/package-browser-field-spec#spec), which in turn create their own packages for client-side use. With a little creativity, this field allows us to map an alias to the `.vue` file itself. For example:
```